src/components/item/ItemForm.vue

```diff
@@ -64,62 +71,79 @@ export default {
   data() {
     return {
       showingStatuses: {
-        editTagsForm: false,
+        detailsTextarea: false,
+        addons: false,
       },
     };
   },
   computed: {
     ...mapGetters([
       'currentListItems',
     ]),
-    ...mapGetters('lists', [
-      'currentListTags',
-      'currentListCategories',
-    ]),
     ...mapGetters('items', [
       'edittingItemIndex',
       'edittingItemObj',
+      'currentItemTags',
     ]),
     ...mapGetters('settings', [
       'isItemFormInSidebar',
     ]),
-    currentItemTags() {
-      return this.currentListTags.filter(
-        listTag => this.edittingItemObj.tags.includes(listTag.id),
-      );
+    itemName() {
+      return this.isUntitled
+        ? ''
+        : this.edittingItemObj.title;
     },
-    isAnyTagWithIdExist() {
-      return this.currentListTags?.some(
-        tag => tag.id,
-      );
+    isUntitled() {
+      return !this.edittingItemObj.title;
     },
-    isAnyCategoryWithIdExist() {
-      return this.currentListCategories?.some(
-        category => category.id,
-      );
+    titlePlaceholder() {
+      let placeholder = '';
+
+      if (this.edittingItemObj.temporaryId) {
+        placeholder = this.$options.NEW_ITEM_PLACEHOLDER;
+      } else if (this.isUntitled) {
+        placeholder = this.$options.UNTITLED_ITEM_PLACEHOLDER;
+      }  
+
+      return placeholder;
     },
-    areTextFieldsEmpty() {
-      return !this.edittingItemObj.title && !this.edittingItemObj.details;
+    isDetailsTextareaShown() {
+      return !!this.edittingItemObj.details.length 
+        || this.showingStatuses.detailsTextarea;
+    },
+    areAddonsShown() {
+      return typeof this.edittingItemObj.category === 'number'
+        || !!this.currentItemTags?.length
+        || !!this.edittingItemObj.relatedItems?.length
+        || !!this.edittingItemObj.relatedLists?.length
+        || this.showingStatuses.addons;
     },
-  },
-  mounted() {
-    if (!this.edittingItemObj?.id) {
-      this.$refs.itemTitle.$el.focus();
-    }
   },
   unmounted() {
-    const emptyItemTitle = '[empty item]';
+    const { title, details, temporaryId } = this.edittingItemObj;
+
+    if (!title && details) {
+      this.updateItemFieldLocally({
+        field: 'title',
+        value: generateTitleFromDetails(details),
+      });
+    }
+
+    this[temporaryId 
+      ? '_addItemOnServer'
+      : '_updateItemOnServer'
+    ]({ item: this.edittingItemObj, cancelToken: null });
```

Comment:
This ternary repeating three times in this pull-request
I think it's better to move somewhere in one place